partner_ref_unique/models/res_partner.py

```diff
@@ -1,18 +1,26 @@
 # Copyright 2016 Antonio Espinosa
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
 
-from odoo import _, api, models
+from odoo import _, api, fields, models
 from odoo.exceptions import ValidationError
 
 
 class ResPartner(models.Model):
     _inherit = "res.partner"
 
+    # This related is needed in order to trigger the check when changing the
+    # value on res.company
+    partner_ref_unique = fields.Selection(
+        related='company_id.partner_ref_unique',
+        store=True,
```

Comment:
Don't store it for performance reason and readonly=True is the default

Reply:
@pedrobaeza I didn't have this commit (https://github.com/odoo/odoo/commit/62056e826a3b99186284cdfecc132cdfd7b1e8d7) in my code base so I post it here FTR.